### ISSUE TITLE
Creating  Service Account  with Restrict with policy with access to a specific bucket is not enforced   

### ISSUE THREAD

**Issue by aviscode** [minio/minio#12930]
Creating  Service Account  with Restrict with a policy with access to a specific bucket is not enforced   
## Expected Behavior

I'm  Creating  Service Account  with Restrict with a policy with access to a specific bucket like this 
the policy is : 

     {
        "Version": "2012-10-17",
        "Statement": [
           {
               "Effect": "Allow",
               "Action": [
                   "s3:*"
               ],
               "Resource": [
                   "arn:aws:s3:::dell*"
               ]
           }
       ]
     }

![image](https://user-images.githubusercontent.com/16151769/129016279-54ab7e22-5a6d-4891-973c-a7867cfb24af.png)

and when I access the service account I can access all buckets not just with the prefix dell*  as u can see in the pic 
![image](https://user-images.githubusercontent.com/16151769/129016545-6f5f78b5-bfe1-4320-b180-a92a15666a30.png)

## Current Behavior

 I can access all buckets not just with the prefix dell*  as need to be based on the policy 

## Steps to Reproduce (for bugs)

1.create a Service Account  with Restrict policy
2. give access just to the specific bucket in the policy 
3. try accessing another bucket and see that we can access them 

## Your Environment

* Version used (`minio --version`): Docker image  minio/minio   latest 
* Server setup and configuration:  docker run -p 9000:9000  -p 9001:9001  --name minio1   -e "MINIO_ROOT_USER=minioadmin"  -e "MINIO_ROOT_PASSWORD=minioadmin"  minio/minio server /data --console-address ":9001"
* Operating System and version (`uname -a`): windows 10

**Comment by harshavardhana** [comment]
@aviscode who is the parent user here ?

**Comment by aviscode** [comment]
I don't understand 
i have an admin user that is creating a new Service Account with Restrict policy to access a specific bucket is not enforced

**Comment by kannappanr** [comment]
Can you paste the output of `mc ls dell1/zoom`?

**Comment by aviscode** [comment]
![image](https://user-images.githubusercontent.com/16151769/129051209-23d6b55e-a32a-4987-9155-77adc4ab5838.png)

**Comment by vadmeste** [comment]
@aviscode can you run this as well ?

`mc admin user svcacct info <alias> service-account-access-key`

**Comment by aviscode** [comment]
![image](https://user-images.githubusercontent.com/16151769/129051807-45ef6a5d-d50c-448c-ad1e-ab41cf69cba0.png)

**Comment by kannappanr** [comment]
`mc admin user svcacct info <alias> <service-account-access-key> --json`

For alias, create one using the admin account and use that, if you do not have one already
For service-account-access-key please use the service account access key you are using to create the alias dell1, as shown in this image pasted by you https://user-images.githubusercontent.com/16151769/129016545-6f5f78b5-bfe1-4320-b180-a92a15666a30.png

**Comment by aviscode** [comment]
![image](https://user-images.githubusercontent.com/16151769/129054531-e81fe31a-1fb3-45f3-8063-022151b866b5.png)

**Comment by kannappanr** [comment]
what about `mc admin user svcacct info dell1 4Y23OO9SE0AH837SF3WM --json`

**Comment by kannappanr** [comment]
can you also paste the output of `mc alias list dell1`

**Comment by aviscode** [comment]
![image](https://user-images.githubusercontent.com/16151769/129055359-be721ce2-10f0-4d50-a1b1-0e0f082c6ff5.png)

**Comment by aviscode** [comment]
![image](https://user-images.githubusercontent.com/16151769/129055417-49e152b3-99cb-47a1-a7ca-cbcecc507860.png)

**Comment by kannappanr** [comment]
@aviscode Can you please create a user first and then login using that user and then create a service account with the same restrictive policy?

**Comment by aviscode** [comment]
do i need to give the user a policy?

**Comment by aviscode** [comment]
now its working good 
but when i make a ls  I'm not getting all buckets with the prefix 
![image](https://user-images.githubusercontent.com/16151769/129059554-917be619-69c3-415b-b1bb-b32374025216.png)

**Comment by kannappanr** [comment]
Yes, with the service account, you need to add another statement, 
```
{
      "Action": [
        "s3:ListAllMyBuckets"
      ],
      "Effect": "Allow",
      "Resource": "arn:aws:s3:::*"
    }
```
This will list all the buckets. The service account either has permissions to display all the buckets or nothing at all, just as AWS S3

**Comment by aviscode** [comment]
not working good at all 
now I can see all other buckets and not just min and mb in other users buckets and see all users data the he policy is not enforced at all **_!!!_**
![image](https://user-images.githubusercontent.com/16151769/129163924-cd08bedb-9f73-42ae-8293-f593cb81852f.png)
![image](https://user-images.githubusercontent.com/16151769/129181976-b1aec607-dd6a-4f52-9d4b-717bbe71892c.png)
![image](https://user-images.githubusercontent.com/16151769/129182124-bdcb6129-6802-4515-8dca-47691c9dd56e.png)

**Comment by aviscode** [comment]
however, the bug in the main account is still there when creating a service account from the main user the policy is not enforced at all

**Comment by harshavardhana** [comment]
Fixed by #12947
